### PR TITLE
[Merged by Bors] - feat: `IsDedekindRing` is `IsDedekindDomain` minus `IsDomain`

### PR DESCRIPTION
**Pull request by Vierkantor**
This PR defines `IsDedekindRing` as `IsDedekindDomain` without the bundled `IsDomain` hypothesis.

Zulip thread: https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Should.20.60IsDedekindDomain.60.20extend.20.60IsDomain.60.3F

---

 - [x] depends on: #6126

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RingTheory/DedekindDomain/Basic.lean [hunk 1]
```diff
@@ -9,14 +9,17 @@ import Mathlib.RingTheory.Polynomial.RationalRoot
...
 
-This file defines the notion of a Dedekind domain (or Dedekind ring),
-as a Noetherian integrally closed commutative ring of Krull dimension at most one.
+This file defines the notion of a Dedekind ring (domain),
+as a Noetherian integrally closed commutative ring (domain) of Krull dimension at most one.
 
 ## Main definitions
 
- - `IsDedekindDomain` defines a Dedekind domain as a commutative ring that is
+ - `IsDedekindRing` defines a Dedekind domain as a commutative ring that is
```
Comment (jcommelin) [inline Mathlib/RingTheory/DedekindDomain/Basic.lean:19]:
```suggestion
 - `IsDedekindRing` defines a Dedekind ring as a commutative ring that is
```

File: Mathlib/RingTheory/DedekindDomain/Basic.lean [hunk 2]
```diff
@@ -90,11 +98,35 @@ theorem DimensionLEOne.eq_bot_of_lt [Ring.DimensionLEOne R] (p P : Ideal R) [p.I
...
+/-- A Dedekind ring is a commutative ring that is Noetherian, integrally closed, and
+has Krull dimension at most one.
+
+This is exactly `IsDedekindDomain` minus the `IsDomain` hypothesis.
+
+The integral closure condition is independent of the choice of field of fractions:
+use `isDedekindRing_iff` to prove `IsDedekindRing` for a given `fraction_map`.
+-/
+class IsDedekindRing
+  extends IsNoetherian A A, DimensionLEOne A, IsIntegrallyClosed A : Prop
```
Comment (jcommelin) [inline Mathlib/RingTheory/DedekindDomain/Basic.lean:110]: Does this work
```suggestion
  extends IsNoetherianRing A, DimensionLEOne A, IsIntegrallyClosed A : Prop
```

File: Mathlib/RingTheory/DedekindDomain/Basic.lean [hunk 3]
```diff
@@ -9,14 +9,17 @@ import Mathlib.RingTheory.Polynomial.RationalRoot
...
+This file defines the notion of a Dedekind ring (domain),
+as a Noetherian integrally closed commutative ring (domain) of Krull dimension at most one.
 
 ## Main definitions
 
- - `IsDedekindDomain` defines a Dedekind domain as a commutative ring that is
+ - `IsDedekindRing` defines a Dedekind domain as a commutative ring that is
+   Noetherian, integrally closed in its field of fractions and has Krull dimension at most one.
+   `isDedekindRing_iff` shows that this does not depend on the choice of field of fractions.
+ - `IsDedekindDomain` defines a Dedekind domain as an integral domain that is
```
Comment (jcommelin) [inline Mathlib/RingTheory/DedekindDomain/Basic.lean:22]: I think you can golf this comment by simply saying that it is a Dedekind ring that is a domain.

File: Mathlib/RingTheory/DedekindDomain/Basic.lean (hunk 2 above)
Comment (Vierkantor) [reply Mathlib/RingTheory/DedekindDomain/Basic.lean:108]: (It did not: `IsNoetherianRing` is not a structure.)

File: Mathlib/RingTheory/DedekindDomain/Basic.lean [hunk 4]
```diff
@@ -90,11 +96,35 @@ theorem DimensionLEOne.eq_bot_of_lt [Ring.DimensionLEOne R] (p P : Ideal R) [p.I
...
+
+This is exactly `IsDedekindDomain` minus the `IsDomain` hypothesis.
+
+The integral closure condition is independent of the choice of field of fractions:
+use `isDedekindRing_iff` to prove `IsDedekindRing` for a given `fraction_map`.
+-/
+class IsDedekindRing
+  extends IsNoetherian A A, DimensionLEOne A, IsIntegrallyClosed A : Prop
+
+/-- An integral domain is a Dedekind domain iff and only if it is
```
Comment (riccardobrasca) [inline Mathlib/RingTheory/DedekindDomain/Basic.lean:110]:
```suggestion
/-- An integral domain is a Dedekind domain if and only if it is
```

File: Mathlib/RingTheory/DedekindDomain/Basic.lean [hunk 5]
```diff
@@ -103,9 +133,18 @@ This is the default implementation, but there are equivalent definitions,
 TODO: Prove that these are actually equivalent definitions.
 -/
 class IsDedekindDomain
-  extends IsDomain A, IsNoetherian A A, DimensionLEOne A, IsIntegrallyClosed A : Prop
+  extends IsDomain A, IsDedekindRing A : Prop
 #align is_dedekind_domain IsDedekindDomain
 
+/-- A Dedekind domain is exactly a Dedekind ring with an extra `IsDomain` hypothesis.
```
Comment (riccardobrasca) [inline Mathlib/RingTheory/DedekindDomain/Basic.lean:139]: This docstring already appears above. Do we want it twice (I have no special opinion on that, just to check this is not an accident)?.

**Review by riccardobrasca — APPROVED** [review]
Thanks!

bors d+

File: Mathlib/RingTheory/DedekindDomain/Basic.lean (hunk 5 above)
Comment (Vierkantor) [reply Mathlib/RingTheory/DedekindDomain/Basic.lean:139]: It is more or less on purpose: I like to start docstrings with a description of the declaration itself, before going to implementation details. How about the following:

```suggestion
/-- Make a Dedekind domain from a Dedekind ring given that it is a domain.
```

**Comment by Vierkantor** [comment]
Since I heard no complaints over my answer to your question, I'm going to go ahead and:

bors merge

:)

**Pull request closed without merging**